### PR TITLE
chore: organizes code in dashboard component

### PR DESCRIPTION
**Pull request by bartoszmajsak**
## Description

This PR extracts certain config/deployment work need for Dashboard to their own funcs for better readability.

## How Has This Been Tested?

`make test` + some manual testing as part of service-mesh work. Hoping for CI job to cover e2e here :)

## Merge criteria:

- [x] The commits are squashed in a cohesive manner and have meaningful messages.
- [x] Testing instructions have been added in the PR body (for PRs involving changes that are not immediately obvious).
- [x] The developer has manually tested the changes and verified that the changes work

### CODE REVIEW

File: components/dashboard/dashboard.go [hunk 1]
```diff
@@ -40,6 +40,9 @@ var imageParamMap = map[string]string{
 	"odh-dashboard-image": "RELATED_IMAGE_ODH_DASHBOARD_IMAGE",
 }
 
+// Verifies that Dashboard implements ComponentInterface
+var _ components.ComponentInterface = (*Dashboard)(nil)
```
Comment (zdtsw) [inline components/dashboard/dashboard.go:44]: hum, duplicated as line89?

Comment (bartoszmajsak) [reply components/dashboard/dashboard.go:44]: Good catch. I blame morning rebase for and GoLand's auto-merge magic for that :)

Comment (bartoszmajsak) [inline components/dashboard/dashboard.go:44]:
```suggestion
```

**Review by VaishnaviHire — APPROVED** [review]
Tested this successfully alongwith #593. 

/lgtm

**Pull request merged**